fotoapparat/src/main/java/io/fotoapparat/parameter/range/IntervalRange.java

```diff
@@ -0,0 +1,58 @@
+package io.fotoapparat.parameter.range;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Implementation of {@link Range} that represents numeric interval.
+ *
+ * @param <T> type of numbers in that interval.
+ */
+public class IntervalRange<T extends Number & Comparable<T>> implements Range<T> {
```

Comment:
I don't have a strong opinion on that, but sure - if you like, you can add a static factory. I see it as a separate `Ranges` class with a static method `range(T lowerBound, T upperBound)`.

In the future, I believe we might have also a `DiscreteRange`.